src/invidious/routes/channels.cr

```diff
@@ -112,6 +112,54 @@ module Invidious::Routes::Channels
     templated "channel"
   end
 
+  def self.podcasts(env)
+    data = self.fetch_basic_information(env)
+    return data if !data.is_a?(Tuple)
+
+    locale, user, subscriptions, continuation, ucid, channel = data
+
+    sort_by = ""
+    sort_options = [] of String
+
+    if channel.auto_generated
+      return env.redirect "/channel/#{channel.ucid}"
+    end
+
+    items, next_continuation = fetch_channel_podcasts(
+      channel.ucid, channel.author, continuation
+    )
+
+    items = items.select(SearchPlaylist).map(&.as(SearchPlaylist))
```

Comment:
You don't need to cast data types. `.select()` already returns an array with a different type (in this case, `Array(SearchPlaylist)`, where the first assignment of `items` was `Array(SearchItem)`

I don't know why it was written like that in the playlists section above...

```suggestion
    items = items.select(SearchPlaylist)
```